docker/Dockerfile.template

```diff
@@ -0,0 +1,32 @@
+# syntax = docker/dockerfile:1
+# Use the official MongoDB image as the base image
+FROM mongo:%%MONGO_VERSION%%
+
+ENV MI_DATABASE_NAME=${MI_DATABASE_NAME:-mongo} \
+    MI_DATABASE_USERNAME=${MI_DATABASE_NAME:-aerius} \
+    MI_DATABASE_PASSWORD=${MI_DATABASE_NAME:-aerius} \
+    MI_DATABASE_VERSION=${MI_DATABASE_VERSION} \
+    MI_INPUT_FILE=${MI_INPUT_FILE} \
+    MI_RUN_SCRIPT_FOLDER=${MI_RUN_SCRIPT_FOLDER} \
+    MI_NEXUS_BASE_URL=${MI_NEXUS_BASE_URL} \
+    MI_NEXUS_REPOSITORY=${MI_NEXUS_REPOSITORY} \
+    MI_BIN_FOLDER=${MI_BIN_FOLDER:-/mongo-initializr} \
+    MI_SOURCE_FOLDER=${MI_SOURCE_FOLDER:-/source} \
+    MI_DBDATA_FOLDER=${MI_DBDATA_FOLDER:-/dbdata} \
+    MI_SKIP_DBDATA_SYNC=${MI_SKIP_DBDATA_SYNC:-false} \
+    MI_BIN_FOLDER_CLEANUP=${MI_BIN_FOLDER_CLEANUP:-false} \
+    MI_SOURCE_FOLDER_CLEANUP=${MI_SOURCE_FOLDER_CLEANUP:-false} \
+    MI_DBDATA_FOLDER_CLEANUP=${MI_DBDATA_FOLDER_CLEANUP:-false} \
+    MI_SKIP_UNSET_ENVS=${MI_SKIP_UNSET_ENVS:-false}
+
+# Install required packages
+RUN apt-get update \
+    && apt-get install -y wget curl jq \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy all necessary scripts
+COPY docker/mongo-initializr.sh /
+COPY docker/docker-entrypoint.sh /
+COPY bin/ "${MI_BIN_FOLDER}"
+
+ENTRYPOINT ["/docker-entrypoint.sh"]
```

Comment:
This overrides the default mongo entrypoint: https://github.com/docker-library/mongo/blob/e3137ef2ab515a4b7b02487ec72cf3a02d14b667/7.0/docker-entrypoint.sh

Which contains some fixes and workarounds and also does stuff like creating default database and user/passwords based on ENV variables. For the sake of compatibility we would need to call that one somewhere unless it breaks some of the things we do.

Our entrypoint seems to point directly to `mongo-initializr.sh`. Technically we could remove ours and just use the as a CMD, or use a RUN construct as done in the example.

Reply:
I don't like depend on the original entrypoint script. I don't fully understand the script and I'm not sure if it can cause any problems.

We should run our script as `ENTRYPOINT`.
In the original base image `mongod` is set as `CMD` what I would like to keep as is.
And the disadvantage of a `RUN` is that it cannot be overwritten like the `ENTRYPOINT`.